code/game/sound.dm

```diff
@@ -40,8 +40,53 @@ ignore_walls - Whether or not the sound can pass through walls.
 falloff_distance - Distance at which falloff begins. Sound is at peak volume (in regards to falloff) aslong as it is in this range.
 
 */
-
-/proc/playsound(atom/source, soundin, vol as num, vary, extrarange as num, falloff_exponent = SOUND_FALLOFF_EXPONENT, frequency = null, channel = 0, pressure_affected = TRUE, ignore_walls = TRUE, falloff_distance = SOUND_DEFAULT_FALLOFF_DISTANCE, use_reverb = TRUE)
+GLOBAL_LIST_INIT(used_sound_channels, list(
+	CHANNEL_LOBBYMUSIC,
+	CHANNEL_ADMIN,
+	CHANNEL_VOX,
+	CHANNEL_JUKEBOX,
+	CHANNEL_HEARTBEAT,
+	CHANNEL_AMBIENT_EFFECTS,
+	CHANNEL_AMBIENT_MUSIC,
+	CHANNEL_BUZZ,
+	CHANNEL_ENGINE_ALERT,
+	CHANNEL_SOUND_EFFECTS,
+	CHANNEL_SOUND_FOOTSTEPS,
+	CHANNEL_WEATHER,
+	CHANNEL_MACHINERY,
+	CHANNEL_INSTRUMENTS,
+	CHANNEL_INSTRUMENTS_ROBOT,
+	CHANNEL_MOB_SOUNDS,
+))
+
+GLOBAL_LIST_INIT(proxy_sound_channels, list(
+	CHANNEL_SOUND_EFFECTS,
+	CHANNEL_SOUND_FOOTSTEPS,
+	CHANNEL_WEATHER,
+	CHANNEL_MACHINERY,
+	CHANNEL_INSTRUMENTS,
+	CHANNEL_INSTRUMENTS_ROBOT,
+	CHANNEL_MOB_SOUNDS,
+))
+
+/proc/guess_mixer_channel(soundin)
```

Comment:
>:(
Using path based guessing is okay i guess but it relys on people having actually sorted sounds, which isn't the case in a fair few locations